packages/fluentui/react-builder/src/components/Canvas.tsx

```diff
@@ -115,6 +119,17 @@ export const Canvas: React.FunctionComponent<CanvasProps> = ({
     [iframeCoordinatesToWindowCoordinates, onMoveComponent],
   );
 
+  const [bodyFocused, setBodyFocused] = React.useState(false);
+  const handleFocus = (ev: FocusEvent) => {
+    if (ev.target && (ev.target as any).getAttribute('data-builder-id') === null) {
+      !bodyFocused && setHeaderMessage('Warning: Focus on body.');
+      !bodyFocused && setBodyFocused(true);
+    } else {
+      bodyFocused && setHeaderMessage('');
+      bodyFocused && setBodyFocused(false);
+    }
+  };
```

Comment:
I think that this could be cleaner and easier to read

```suggestion
  const handleFocus = (ev: FocusEvent) => {
    // can add a comment here of what this check means in reality
    if (!ev.target || ev.target.getAttribute('data-builder-id')) {
      return;
    }
    
    if (bodyFocused) {
      setHeaderMessage('.');
      setBodyFocused(false);
    } else {
      setHeaderMessage('Warning: Focus on body.');
      setBodyFocused(true);
    }
```